src/assets/assetMetrics.spec.js

```diff
@@ -0,0 +1,476 @@
+      let valuesFromServerAfterFormat;
+      let valuesFromServerBeforeFormat;
+
+      beforeEach(function() {
+        numberOfAssetMetrics = faker.random.number({ min: 1, max: 10 });
+        assetTypeId = fixture.build('assetType').id;
+        paginationOptionsBeforeFormat = {
+          limit: faker.random.number({ min: 10, max: 1000 }),
+          offset: faker.random.number({ max: 1000 })
+        };
+        paginationOptionsAfterFormat = {
+          limit: faker.random.number({ min: 10, max: 1000 }),
+          offset: faker.random.number({ max: 1000 })
+        };
+        valuesFromServerAfterFormat = {
+          _metadata: fixture.build('paginationMetadata'),
+          records: fixture.buildList('assetMetric', numberOfAssetMetrics, {
+            assetTypeId
+          })
+        };
+        valuesFromServerBeforeFormat = {
+          ...valuesFromServerAfterFormat,
+          records: valuesFromServerAfterFormat.records.map((values) =>
+            fixture.build('assetMetric', values, { fromServer: true })
+          )
+        };
+
+        formatPaginatedDataFromServer = this.sandbox
+          .stub(paginationUtils, 'formatPaginatedDataFromServer')
+          .returns(valuesFromServerAfterFormat);
+        request = {
+          ...baseRequest,
+          get: this.sandbox.stub().resolves(valuesFromServerBeforeFormat)
+        };
+        toSnakeCase = this.sandbox
+          .stub(objectUtils, 'toSnakeCase')
+          .returns(paginationOptionsAfterFormat);
+
+        const assetMetrics = new AssetMetrics(baseSdk, request, expectedHost);
+        promise = assetMetrics.getAll(
+          assetTypeId,
+          paginationOptionsBeforeFormat
+        );
+      });
+
+      it('formats the pagination options sent to the server', function() {
+        expect(toSnakeCase).to.be.calledWith(paginationOptionsBeforeFormat);
+      });
+
+      it('gets a list of the asset metrics from the server', function() {
+        expect(request.get).to.be.calledWith(
+          `${expectedHost}/assets/types/${assetTypeId}/metrics`,
+          { params: paginationOptionsAfterFormat }
+        );
+      });
+
+      it('formats the asset metric data', function() {
+        return promise.then(() => {
+          expect(formatPaginatedDataFromServer).to.be.calledWith(
+            valuesFromServerBeforeFormat
+          );
+        });
+      });
+
+      it('resolves with a list of asset metrics', function() {
+        return expect(promise).to.be.fulfilled.and.to.eventually.deep.equal(
+          valuesFromServerAfterFormat
+        );
+      });
+    });
+
+    context('when there is missing required information', function() {
+      let promise;
+
+      beforeEach(function() {
+        const assetMetrics = new AssetMetrics(
+          baseSdk,
+          baseRequest,
+          expectedHost
+        );
+
+        promise = assetMetrics.getAll();
+      });
+      it('throws an error when the asset type ID is missing', function() {
```

Comment:
Can you add a blank line before this `it`

Reply:
I'm on it!